VCC_Projekt/Components/Pages/EditRoles.razor.cs

```diff
@@ -0,0 +1,152 @@
+                .Join(
+                dbContext.Users,
+                ur => ur.UserId,
+                u => u.Id,
+                (ur, u) => new User
+                (
+                    u.UserName,
+                    u.Firstname,
+                    u.Lastname,
+                    u.Email,
+                    ur.RoleId
+                )
+                )
+                .ToList();
+            roles = dbContext.Roles.Where(r => r.Name != "Benutzer").Select(r => r.Name).ToList();
+        }
+
+        private async Task<IEnumerable<User>> SearchUsers(string searchText, CancellationToken cancellationToken)
+        {
+            if (string.IsNullOrEmpty(searchText)) return Enumerable.Empty<User>(); 
+
+            string tempSearchtext = searchText.ToUpper();
+
+            return await dbContext.Users
+                        .AsNoTracking()
+                        .Where(u => u.NormalizedEmail.Contains(tempSearchtext) ||
+                                    u.NormalizedUserName.Contains(tempSearchtext) ||
+                                    u.Firstname.ToUpper().Contains(tempSearchtext) ||
+                                    u.Lastname.ToUpper().Contains(tempSearchtext))
+                        .Select(u => new User(u.UserName, u.Firstname, u.Lastname, u.Email, null))
+                        .ToListAsync(cancellationToken);
+        }
+
+        private async Task AddRole()
+        {
+            showErrorAlert = false;
+            showSuccessAlert = false;
+            var user = await Usermanager.FindByEmailAsync(modelUser.User.Email);
+            if (!await Usermanager.IsInRoleAsync(user, modelUser.Role))
+            {
+                var result = await Usermanager.AddToRoleAsync(user, modelUser.Role);
+                if (result.Succeeded)
+                {
+                    showSuccessAlert = true;
+                    successMessage = "Rolle wurde hinzugefügt";
+                }
+                else
+                {
+                    showErrorAlert = true;
+                    errorMessage = "Fehler beim hinzufügen der Rolle";
+                }
+            }
+            else
+            {
+                showErrorAlert = true;
+                errorMessage = "Rolle für Benutzer existiert schon!";
+            }
+            OnInitialized();
+            StateHasChanged();
+
+        }
+
+        private async Task RemoveRole(User user)
+        {
+            var userResult = await Usermanager.FindByEmailAsync(user.Email);
+            await Usermanager.RemoveFromRoleAsync(userResult,user.Rolename);
+            showSuccessAlert = true ;
+            successMessage = $"Rolle wurde entzogen ({user.ToString()})";
+            OnInitialized();
+            StateHasChanged();
+        }
+
+        private Func<User, bool> _quickFilter => x =>
+        {
+            if (string.IsNullOrWhiteSpace(_searchString))
+                return true;
```

Comment:
Am anfang sollte nichts in der Tabelle sein.
```suggestion
                return false;
```